src/commandsAndMenu.tsx

```diff
@@ -479,24 +480,53 @@ export function addCommands(
               tooltip: trans.__('Refresh diff widget'),
               className: 'jp-git-diff-refresh'
             });
-            refreshButton.hide();
-            diffWidget.toolbar.addItem('refresh', refreshButton);
 
-            const refresh = () => {
-              refreshButton.show();
-            };
-
-            model.changed.connect(refresh);
-            widget.disposed.connect(() => {
-              model.changed.disconnect(refresh);
+            const resolveButton = new ToolbarButton({
```

Comment:
Could you create the buttons in the `if/else` block? So we don't create them if they are not needed.